src/Symfony/Component/DependencyInjection/ContainerBuilder.php

```diff
@@ -1114,6 +1121,61 @@ public function getEnvCounters()
         return $this->envCounters;
     }
 
+    private function addOverriddenGetters($id, Definition $definition, \ReflectionClass $class)
+    {
+        if (PHP_VERSION_ID < 70000) {
+            throw new RuntimeException(sprintf('Getter-based injection requires PHP 7 or more as used for service "%s".', $id));
+        }
+        if ($class->isFinal()) {
+            throw new RuntimeException(sprintf('Invalid getter for service "%s": class "%s" is final.', $id, $class->name));
+        }
+        $getters = '';
+        foreach ($definition->getOverriddenGetters() as $name => $returnValue) {
+            if (!$class->hasMethod($name)) {
+                throw new RuntimeException(sprintf('Invalid getter for service "%s": method "%s::%s" does not exist.', $id, $class->name, $name));
+            }
+            $r = $class->getMethod($name);
+            if ($r->isPrivate()) {
+                throw new RuntimeException(sprintf('Invalid getter for service "%s": method "%s::%s" is private.', $id, $class->name, $name));
+            }
+            if ($r->isFinal()) {
+                throw new RuntimeException(sprintf('Invalid getter for service "%s": method "%s::%s" is final.', $id, $class->name, $name));
+            }
+            if ($r->isInternal()) {
+                throw new RuntimeException(sprintf('Invalid getter for service "%s": method "%s::%s" is internal.', $id, $class->name, $name));
+            }
+            if ($r->returnsReference()) {
+                throw new RuntimeException(sprintf('Invalid getter for service "%s": method "%s::%s" returns by reference.', $id, $class->name, $name));
+            }
+            if (0 < $r->getNumberOfParameters()) {
+                throw new RuntimeException(sprintf('Invalid getter for service "%s": method "%s::%s" has parameters.', $id, $class->name, $name));
+            }
+            if ($type = $r->getReturnType()) {
+                $type = ': '.($type->allowsNull() ? '?' : '').$this->generateTypeHint($type, $r);
```

Comment:
Indeed, and this code is fine on php7.0 also because of the above "if". On 7.0 a return type can't be nullable, but the method exists. So: all is fine :)